### PR TITLE
Solution_py_distance_class

### PR DESCRIPTION
**Pull request by VladKrit-commits**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,62 @@
...
+
+    def __eq__(self, other: int | float | Distance) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km == other
+        return self.km == other.km
+
+    def __le__(self, other: int | float | Distance) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km <= other
+        return self.km <= other.km
```
Comment (Dimosphen1) [inline app/main.py:57]: You could rely on the implementation of other methods such as `__gt__` here and in other methods as well.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested.

**Comment by VladKrit-commits** [comment]
I made it(i hope so:)) in the "...(changed)" commit

**Review by viktoria-rybenchuk — COMMENTED** [review]
Good job!

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,62 @@
...
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        if isinstance(other, int | float):
+            return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, int | float):
+            return Distance(round(self.km / other, 2))
+        elif isinstance:
```
Comment (pavlejviki) [inline app/main.py:33]:
```suggestion
        elif isinstance(other, Distance):
```

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,62 @@
...
+            raise TypeError(
+                "'__truediv__' method shouldn't accept Distance instance"
+            )
+        return Distance(self.km / other.km)
```
Comment (pavlejviki) [inline app/main.py:37]: other can't be Distance instance so this return is redundant

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,62 @@
...
+        if isinstance(other, int | float):
+            return self.km > other
+        return self.km > other.km
+
+    def __eq__(self, other: int | float | Distance) -> bool:
+        if isinstance(other, int | float):
+            return self.km == other
+        return self.km == other.km
+
+    def __le__(self, other: int | float | Distance) -> bool:
```
Comment (pavlejviki) [inline app/main.py:54]: You can use previously created magic method here(and in ge as well) to simplify your code. e.g. return not self > other

**Review by pavlejviki — CHANGES_REQUESTED** [review]

**Comment by VladKrit-commits** [comment]
Done in solution(changed_2)

**Pull request closed without merging**